### PR TITLE
MINOR: [C++] Row wise conversion example fix

### PR DESCRIPTION
**Pull request by ChaomingZhangCN**
This example should print the expected rows.

### CODE REVIEW

**Comment by pitrou** [comment]
Can you explain why it should print the expected rows?

**Comment by ZhangChaoming** [comment]
@lidavidm Yep. Some check were not successful, but I think my changes is safe, could you please help merge it?

**Comment by pitrou** [comment]
Hmm... can we instead avoid flipping `rows` and `expected_rows` to make the example less confusing?

**Comment by ZhangChaoming** [comment]
> Hmm... can we instead avoid flipping `rows` and `expected_rows` to make the example less confusing?

I did not get it, you mean just use one variable `rows` ?

**Comment by lidavidm** [comment]
Swap the variable names.

**Comment by ZhangChaoming** [comment]
@pitrou DONE

**Review by pitrou — APPROVED** [review]

File: cpp/examples/arrow/row_wise_conversion_example.cc [hunk 1]
```diff
@@ -172,17 +172,19 @@ arrow::Result<std::vector<data_row>> ColumnarTableToVector(
...
 }
 
+
 arrow::Status RunRowConversion() {
-  std::vector<data_row> rows = {
+  std::vector<data_row> original_rows = {
       {1, 1, {10.0}}, {2, 3, {11.0, 12.0, 13.0}}, {3, 2, {15.0, 25.0}}};
   std::shared_ptr<arrow::Table> table;
-  std::vector<data_row> expected_rows;
+  std::vector<data_row> converted_rows,
```
Comment (pitrou) [inline cpp/examples/arrow/row_wise_conversion_example.cc:180]:
```suggestion
  std::vector<data_row> converted_rows;
```

**Pull request merged**